PowerSimulationsPlots/src/call_plots.jl

```diff
@@ -193,47 +228,57 @@ function bar_plot(res::PSI.Results; kwargs...)
     save_fig = get(kwargs, :save, nothing)
     bar_gen = get_bar_gen_data(res)
     if isnothing(backend)
-        @info("No backend detected. Setting GR() by default.")
-        Plots.gr()
-    end
-    if backend == Plots.PlotlyJSBackend()
-        seriescolor = get(kwargs, :seriescolor, plotly_default)
-        plotly_bar_plots(res, seriescolor; kwargs...)
-        plotly_bar_gen(bar_gen, seriescolor; kwargs...)
-    else
-        seriescolor = get(kwargs, :seriescolor, gr_default)
-        for name in string.(keys(res.variables))
-            variable_bar = get_bar_plot_data(res, name)
-            p = RecipesBase.plot(variable_bar, name; seriescolor = seriescolor)
-            set_display && display(p)
-            !isnothing(save_fig) && Plots.savefig(p, joinpath(save_fig, "$(name)_Bar.png"))
-        end
-        p2 = RecipesBase.plot(bar_gen; seriescolor = seriescolor)
-        set_display && display(p2)
-        !isnothing(save_fig) && Plots.savefig(p2, joinpath(save_fig, "Bar_Generation.png"))
+        throw(IS.ConflictingInputsError("No backend detected. Type gr() to set a backend."))
     end
+    _bar_plot_internal(res, bar_gen, backend, save_fig, set_display; kwargs...)
 end
 
-function bar_plot(res::PSI.Results, variables::Array; kwargs...)
-    res_var = Dict()
-    for variable in variables
-        res_var[variable] = res.variables[variable]
+function _bar_plot_internal(
+    res::PSI.Results,
+    bar_gen::BarGeneration,
+    backend::Plots.PlotlyJSBackend,
+    save_fig::Any,
+    set_display::Bool;
+    kwargs...,
+)
+    seriescolor = get(kwargs, :seriescolor, plotly_default)
+    plotly_bar_plots(res, seriescolor; kwargs...)
+    plotly_bar_gen(bar_gen, seriescolor; kwargs...)
+end
+
+function _bar_plot_internal(
+    res::PSI.Results,
+    bar_gen::BarGeneration,
+    backend::Any,
+    save_fig::Any,
+    set_display::Bool;
+    kwargs...,
+)
+    seriescolor = get(kwargs, :seriescolor, gr_default)
+    for name in string.(keys(res.variables))
+        variable_bar = get_bar_plot_data(res, name)
+        p = RecipesBase.plot(variable_bar, name; seriescolor = seriescolor)
+        set_display && display(p)
+        if !isnothing(save_fig)
```

Comment:
After the change above:
```suggestion
        if save_fig
```

And then there are several more instances of this.